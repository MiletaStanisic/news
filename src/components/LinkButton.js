import React from 'react'
import { Link } from 'react-router-dom';

function LinkButton({ onClick, to, text }) {
  return (
    <Link
      onClick={onClick}
      to={to}
      className="btn btn-link card-link">{text}
    </Link>
  )
}


export default LinkButton;
