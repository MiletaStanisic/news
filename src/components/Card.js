import React from 'react'
import { useDispatch } from 'react-redux';
import { loadArticle } from '../store/actions';

import defaultImage from '../assets/images/news-default.jpeg';
import LinkButton from './LinkButton';

function Card({ article }) {
  const dispatch = useDispatch();

  return (
    <div className="card mb-4">
      <img src={article.urlToImage || defaultImage} className="card-img-top cover" alt="Cover" />
      <div className="card-body">
        <h5 className="card-title line-clamp">{article.title}</h5>
        <p className="card-text line-clamp">{article.description}</p>
      </div>
      <div className="card-body">
        <LinkButton
          onClick={() => dispatch(loadArticle(article))}
          to={`article/${encodeURIComponent(article.title)}`}
          text="Read more" />
      </div>
    </div>
  )
}

export default Card;
