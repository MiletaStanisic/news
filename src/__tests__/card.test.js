import * as React from 'react';
import { shallow } from 'enzyme';
import Card from '../components/Card';
import LinkButton from '../components/LinkButton';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(fn => fn()),
  useDispatch: jest.fn()
}));


describe('<Card />', () => {
  const wrapper = shallow(<Card article={{
    title: 'Test title',
    description: 'Lorem ipsum',
    urlToImage: 'https://example.com',
  }} />);

  it('renders card correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('card'));
  });

  it('Click event', () => {
    const mockCallBack = jest.fn();

    const button = shallow((<LinkButton onClick={mockCallBack} to={'article/id'} text="Read more" />));
    button.find('.btn').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});