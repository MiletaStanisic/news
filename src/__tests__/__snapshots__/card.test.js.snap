// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders card correctly 1`] = `ShallowWrapper {}`;
